android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/UserLocationView.java

```diff
@@ -414,6 +410,11 @@ public void onSensorChanged(SensorEvent event) {
                 return;
             }
 
+            long currentTime = System.currentTimeMillis();
```

Comment:
I think it would be better to avoid using `System.currentTimeMillis()` and instead use [SystemClock](http://developer.android.com/reference/android/os/SystemClock.html) `elapsedTime` method.

From the `SystemClock` javadoc: 
>System.currentTimeMillis() is the standard "wall" clock (time and date) expressing milliseconds since the epoch. The wall clock can be set by the user or the phone network (see setCurrentTimeMillis(long)), so the time may jump backwards or forwards unpredictably.